components/addressmanager/src/lib.rs

```diff
@@ -2,94 +2,82 @@ mod stores;
+
+pub use stores::NetAddress;
 
 const MAX_ADDRESSES: usize = 4096;
 const MAX_CONNECTION_FAILED_COUNT: u64 = 3;
 
 pub struct AddressManager {
     banned_address_store: DbBannedAddressesStore,
-    not_banned_address_store: not_banned_address_store_with_cache::Store,
-}
-
-#[derive(PartialEq, Eq, Hash, Copy, Clone)]
-pub struct NetAddress {
-    ip: Ipv6Addr,
-    port: u16,
-}
-
-impl NetAddress {
-    pub fn new(ip: Ipv6Addr, port: u16) -> Self {
-        Self { ip, port }
-    }
+    address_store: address_store_with_cache::Store,
 }
 
 impl AddressManager {
     pub fn new(db: Arc<DB>) -> Arc<Mutex<Self>> {
         Arc::new(Mutex::new(Self {
             banned_address_store: DbBannedAddressesStore::new(db.clone(), MAX_ADDRESSES as u64),
-            not_banned_address_store: not_banned_address_store_with_cache::new(db),
+            address_store: address_store_with_cache::new(db),
         }))
     }
 
     pub fn add_address(&mut self, address: NetAddress) {
         // TODO: Don't add non routable addresses
 
-        if self.not_banned_address_store.has(address) {
+        if self.address_store.has(address) {
             return;
         }
 
         // We mark `connection_failed_count` as 0 only after first success
-        self.not_banned_address_store.set(address, ConnectionFailureCount(1));
+        self.address_store.set(address, 1);
     }
 
     pub fn mark_connection_failure(&mut self, address: NetAddress) {
-        if !self.not_banned_address_store.has(address) {
+        if !self.address_store.has(address) {
             return;
         }
 
-        let new_count = self.not_banned_address_store.get(address).0 + 1;
+        let new_count = self.address_store.get(address).connection_failed_count + 1;
         if new_count > MAX_CONNECTION_FAILED_COUNT {
-            self.not_banned_address_store.remove(address);
+            self.address_store.remove(address);
         } else {
-            self.not_banned_address_store.set(address, ConnectionFailureCount(new_count));
+            self.address_store.set(address, new_count);
         }
     }
 
     pub fn mark_connection_success(&mut self, address: NetAddress) {
-        if !self.not_banned_address_store.has(address) {
+        if !self.address_store.has(address) {
             return;
         }
 
-        self.not_banned_address_store.set(address, ConnectionFailureCount(0));
+        self.address_store.set(address, 0);
     }
 
     pub fn get_all_addresses(&self) -> impl Iterator<Item = NetAddress> + '_ {
-        self.not_banned_address_store.get_all_addresses()
+        self.address_store.get_all_addresses()
     }
 
-    pub fn get_random_addresses(&self, count: usize, exceptions: HashSet<NetAddress>) -> Vec<NetAddress> {
-        self.not_banned_address_store.get_random_addresses(count, exceptions)
+    pub fn get_random_addresses(&self, exceptions: HashSet<NetAddress>) -> impl Iterator<Item = NetAddress> {
```

Comment:
perhaps call it `iterate_prioritized_random_addresses` (prioritized is optional)